### ISSUE TITLE
shell.exec() ignores fatal option when not set globally

### ISSUE THREAD

**Issue by chrisromano** [shelljs/shelljs#981]
### Node version (or tell us if you're using electron or some other framework):
v10.17.0
### ShellJS version (the most recent version/Github branch you see the bug on):
0.8.3
### Operating system:
Ubuntu 18.04.3 LTS (via Windows 10 WSL)
### Description of the bug:
When running shell.exec(), passing in {fatal:true} option is ignored. 
Other options seem to work correctly, and setting config.fatal globally has desired behavior.

### Example ShellJS command to reproduce the error:
```
# Executing an invalid command
> shelljs.exec('foobar')
/bin/sh: 1: foobar: not found
{ [String: '']
  stdout: '',
  stderr: '/bin/sh: 1: foobar: not found\n',
  code: 127,
  cat: [Function: bound ],
  exec: [Function: bound ],
  grep: [Function: bound ],
  head: [Function: bound ],
  sed: [Function: bound ],
  sort: [Function: bound ],
  tail: [Function: bound ],
  to: [Function: bound ],
  toEnd: [Function: bound ],
  uniq: [Function: bound ] }

# Same behavioir when passing in fatal:true
> shelljs.exec('foobar', {fatal:true})
/bin/sh: 1: foobar: not found
{ [String: '']
  stdout: '',
  stderr: '/bin/sh: 1: foobar: not found\n',
  code: 127,
  cat: [Function: bound ],
  exec: [Function: bound ],
  grep: [Function: bound ],
  head: [Function: bound ],
  sed: [Function: bound ],
  sort: [Function: bound ],
  tail: [Function: bound ],
  to: [Function: bound ],
  toEnd: [Function: bound ],
  uniq: [Function: bound ] }

# Setting global option has expected behavior
> shelljs.config.fatal = true
true
> shelljs.exec('foobar')
/bin/sh: 1: foobar: not found
Thrown:
Error: exec: /bin/sh: 1: foobar: not found

    at Object.error (/home/chris/sample/node_modules/shelljs/src/common.js:110:27)
    at execSync (/home/chris/sample/node_modules/shelljs/src/exec.js:102:12)
    at Object._exec (/home/chris/sample/node_modules/shelljs/src/exec.js:205:12)
    at Object.exec (/home/chris/sample/node_modules/shelljs/src/common.js:335:23)

# Setting silent: true appears to have desired effect.
> shelljs.exec('foobar', {silent:true})
Thrown:
Error: exec: /bin/sh: 1: foobar: not found

    at Object.error (/home/chris/sample/node_modules/shelljs/src/common.js:110:27)
    at execSync (/home/chris/sample/node_modules/shelljs/src/exec.js:102:12)
    at Object._exec (/home/chris/sample/node_modules/shelljs/src/exec.js:205:12)
    at Object.exec (/home/chris/sample/node_modules/shelljs/src/common.js:335:23)

**Comment by nfischer** [comment]
This was implemented recently in #961. This hasn't been released yet.

**Comment by HosseinAgha** [comment]
@nfischer this happens for me even if I set fatal option globally. `ShellJS version: ^0.8.4`.

**Comment by nfischer** [comment]
This will be in v0.9.0

**Comment by philippotto** [comment]
> This will be in v0.9.0

This issue still exists in v0.8.5 (which is the newest version). for me (also see the other PRs which are referencing this issue). Any chance there was a regression or the fix didn't get into the release?

**Comment by nfischer** [comment]
The fix is scheduled for v0.9.0, not for v0.8.5. v0.9.0 is still under development. You can monitor progress at [the GitHub milestone tracker](https://github.com/shelljs/shelljs/milestone/6).

v0.8.5 was a very small patch release for a specific issue (you can check the release notes for details). Unfortunately, I wasn't able to cherrypick all the bug patches into that release because some other (breaking) changes have landed on the main branch and it's not trivial to figure out all the changes which are safe to cherrypick for a patch release.

**Comment by chriscarpenter12** [comment]
Just ran into the same issue. If fatal wasn't set globally it was ignored when set on the individual exec command. Specifically I was trying to set globally fatal to `true` and `fatal: false` per exec command it didn't work
